core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -301,6 +301,28 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
     if (proxyUser != null && principal != null) {
       error("Only one of --proxy-user or --principal can be provided.")
     }
+
+    // Ensure heartbeat arguments is valid.
+    val networkTimeout = Utils.timeStringAsSeconds(
+      sparkProperties.getOrElse("spark.network.timeout", "120s"))
+    val executorTimeoutMs =
+      Utils.timeStringAsMs(sparkProperties.getOrElse(
+        "spark.storage.blockManagerSlaveTimeoutMs", s"${networkTimeout}s"))
+    val executorHeartbeatIntervalMs = Utils.timeStringAsMs(
+      sparkProperties.getOrElse("spark.executor.heartbeatInterval", "10s"))
+    val blockManagerTimeoutInterval = Utils.timeStringAsMs(
+      sparkProperties.getOrElse("spark.storage.blockManagerTimeoutIntervalMs", "60s"))
+    val checkNetworkTimeoutIntervalMs =
+      Utils.timeStringAsSeconds(sparkProperties.getOrElse(
+        "spark.network.timeoutInterval", s"${blockManagerTimeoutInterval}ms")) * 1000
```

Comment:
I do not get this change. You have specified the default value for `spark.network.timeoutInterval` as `s"${blockManagerTimeoutInterval}ms"` so if spark.network.timeoutInterval is not set, the value of blockManagerTimeoutInterval which is in ms will be interpreted as seconds. Am I right?

Reply:
@pgandhi999  timeStringAsSeconds will convert the string of s"${blockManagerTimeoutInterval}ms"  to seconds, it will take the units conversion into account.